src/Dialogs/Cloud/WeGlideUploadDialog.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "WeGlideUploadDialog.hpp"
+#include "Interface.hpp"
+#include "Cloud/weglide/WeGlideSettings.hpp"
+#include "Cloud/weglide/UploadFlight.hpp"
+#include "Operation/PluggableOperationEnvironment.hpp"
+#include "net/http/Init.hpp"
+#include "json/ParserOutputStream.hxx"
+#include "Dialogs/CoDialog.hpp"
+#include "Dialogs/Error.hpp"
+#include "UIGlobals.hpp"
+#include "Language/Language.hpp"
+#include "co/InvokeTask.hxx"
+#include "Dialogs/Message.hpp"
+#include "util/Exception.hxx"
+#include "util/ConvertString.hpp"
+#include "io/StringConverter.hpp"
+#include "LogFile.hpp"
+#include "json/Lookup.hxx"
+#include "system/Path.hpp"
+#include "LocalPath.hpp"
+#include "system/FileUtil.hpp"
+
+#include <inttypes.h>
+
+namespace WeGlide {
+
+  typedef struct _Glider {
```

Comment:
Why `typedef`, and what use is the `_Glider` name?

Reply:
In my working environment we use this construct with typedef very often - and I know, that I can use the same name for struct like for the typdef (`typdef struct S S;`), but I learned with your hint from https://www.kernel.org/doc/html/latest/process/coding-style.html#typedefs, that is a 'mistake' in the linux kernel cofig style...
I will change that!

Reply:
The mistake is that you apply an arbitray project's C coding style to C++. In C++, these typedefs have no use (and `typedef` is even a deprecated keyword in C++).

Reply:
Ok, that's new for me!